runtimes/common/src/dip/did.rs

```diff
@@ -98,13 +97,17 @@ where
 		} else {
 			Ok(None)
 		}?;
-		let linked_accounts: Vec<_> =
-			pallet_did_lookup::ConnectedAccounts::<Runtime>::iter_key_prefix(identifier).collect();
 
-		ensure!(
-			<Self as IdentityProvider<Runtime>>::MAX_LINKED_ACCOUNT > linked_accounts.len().saturated_into(),
-			LinkedDidInfoProviderError::MaxLinkedAccounts
-		);
+		let has_user_to_many_acc = pallet_did_lookup::ConnectedAccounts::<Runtime>::iter_key_prefix(identifier)
+			.skip(MAX_LINKED_ACCOUNTS.saturated_into())
+			.next()
+			.is_some();
+
+		ensure!(has_user_to_many_acc, LinkedDidInfoProviderError::MaxLinkedAccounts);
+
+		let linked_accounts: Vec<_> = pallet_did_lookup::ConnectedAccounts::<Runtime>::iter_key_prefix(identifier)
+			.take(MAX_LINKED_ACCOUNTS.saturated_into())
+			.collect();
```

Comment:
Liked this more then: 

```
let linked_accounts: Vec<_> = pallet_did_lookup::ConnectedAccounts::<Runtime>::iter_key_prefix(identifier)
    .enumerate()
    .take(MAX_LINKED_ACCOUNTS.saturated_into())
    .inspect(|(index, _)| {
        if *index == MAX_LINKED_ACCOUNTS.saturated_into() - 1 {
            // Check if there are more accounts beyond the limit
            let has_user_too_many_acc = pallet_did_lookup::ConnectedAccounts::<Runtime>::iter_key_prefix(identifier)
                .skip(MAX_LINKED_ACCOUNTS.saturated_into())
                .next()
                .is_some();

            ensure!(!has_user_too_many_acc, LinkedDidInfoProviderError::MaxLinkedAccounts);
        }
    })
    .map(|(_, account)| account)
    .collect();
```

bit less performant but easier to read

Reply:
Ah, sorry. Little typo. I also think the second option is less easy to read, that's why I chose the first option. I updated the error in [42414a5](https://github.com/KILTprotocol/kilt-node/pull/585/commits/42414a56dc188038a0aac00e6f050f082f2bc7c9)